Noetic support

martiniil opened the pull request:
## Description

These changes add noetic support.

**!!! This pull request must not be merged. It is only created for review.**

## Next steps:

* [x] Review
* [x] Squash into meaningful commits
* [x] Noetic release of `pilz_common`
* [x] Noetic release of `pilz_robots`
* [x] Check CI
* [x] Change default branch
* [x] #341, https://github.com/PilzDE/industrial_ci_addons/pull/7
* [x] Perform noetic release
* [ ] #332, #333

### Things to add, update or check by the maintainers before this PR can be closed.

* [x] Public api function documentation
* [x] Architecture documentation reflects actual code structure
* [x] ~~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~~ postponed
* [x] ~~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~~ postponed
* [x] Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))
* [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
* [x] CHANGELOG.rst updated
* [x] Copyright headers
* [x] Examples

### Review Checklist
* [x] Soft- and hardware architecture (diagrams and description)
* [x] Test review (test plan and individual test cases)
* [x] Documentation describes purpose of file(s) and responsibilities
* [x] Code (coding rules, style guide)

### Release planning (please answer)
* [x] When is the new feature released? asap
* [x] Which dependent packages do have to be released simultaneously? none

martiniil commented:
Tested on ubuntu 20: coverage is now 100%

agutenkunst commented:
CI is stuck?

martiniil commented:
> 
> 
> CI is stuck?

We have to ignore CI for the moment (until the moveit noetic release).

rfeistenauer commented on pilz_robot_programming/test/acceptance_tests/acceptance_test_blending.py:
```diff
@@ -19,6 +19,7 @@
 from geometry_msgs.msg import Pose, Point
 from pilz_robot_programming.robot import *
 from pilz_robot_programming.commands import *
+from pilz_industrial_motion_testutils.acceptance_test_utils import _askPermission, _askSuccess
```
we import private methods?

rfeistenauer commented on pilz_robot_programming/src/pilz_robot_programming/robot.py:
```diff
@@ -187,8 +188,8 @@ def get_current_joint_states(self, planning_group=_DEFAULT_PLANNING_GROUP):
         except MoveItCommanderException as e:
-            rospy.logerr(e.message)
-            raise RobotCurrentStateError(e.message)
+            rospy.logerr(str(e))
+            raise RobotCurrentStateError(str(e))
```
In line 209 its written as "%s" % e and RobotCurrentStateError(e)
We choose one alternative

martiniil committed:
Move pilz command planner to moveit

martiniil committed:
Use public acceptance-test utility methods

martiniil committed:
Update maintainer list

martiniil committed:
Fix KeyError on calling release twice

martiniil committed:
Port to ROS Noetic (ubuntu 20.04, python3)

* Remove pilz_store_positions package
* Update branching model in README.md
* Use relative paths for test-data/movecmd.py (colcon support)
* Misc minor refactorings

martiniil committed:
Adapt to changes in MoveThread api

martiniil committed:
Update changelog

martiniil committed:
Translate german notes in state diagram

martiniil committed:
Fix broken link to brake-test architecture

martiniil committed:
Prevent misunderstandings in acceptance-tests (#330)

* Reformulate some hints
* Fix typo
* Fix indentation in askSuccess calls

Co-authored-by: jschleicher <j.schleicher@pilz.de>

rfeistenauer committed:
Fix workflow definition

martiniil commented:
The latest moveit releases should make prbt_pg70_support available in testing. We just have to wait ...

agutenkunst commented:
@martiniil should work now? http://repositories.ros.org/status_page/ros_noetic_default.html?q=pg70

martiniil committed:
Adjust blend radius in test data (#341)

The blend radius should be strictly less than the distance between start- and endpoint of the LIN commands.

rfeistenauer committed:
Update Changelogs

rfeistenauer committed:
0.5.0

martiniil committed:
Add noetic buildfarm status to README.md (#342)
